### ISSUE TITLE
[P0] Allow tile issue `COMPLETE` signals on ring (within one CGRA) towards controller

### ISSUE THREAD

**Issue by tancheng** [tancheng/VectorCGRA#48]
A program/DFG normally completes once `return` is executed. We need to leverage https://github.com/tancheng/VectorCGRA/blob/master/fu/single/RetRTL.py to notify the controller (then eventually notify CPU) that the execution is done.

But before the above, we can simply check the completeness of a kernel based on:
https://github.com/tancheng/VectorCGRA/blob/ecae2ed31cf567267bbf286e3f5729a55adc79a0/mem/ctrl/CtrlMemDynamicRTL.py#L108
as the `CtrlMemoryDynamicallyRTL` already connects to the `ctrl_ring`, though it can only passively receive `CMD`, so we need to extend it to make it be able to send `CMD`.

In summary:
 - [x] Extend `CtrlMemoryDynamicallyRTL` to make it be able to send `CMD` towards controller.
 - [ ] Extend `RetRTL` to make it be able to send `CMD` towards controller.

Hi @yuqisun, I am assigning this to you as your P0. https://github.com/tancheng/CGRA-Flow/pull/50 also has high priority (say P0.5). Thanks!

**Comment by tancheng** [comment]
This is also a blocker for https://github.com/tancheng/VectorCGRA/issues/91, otherwise, we don't know when the output data is ready in SRAM. (though we can waiting for 1000 cycles to issue a read request from CPU...)

**Comment by tancheng** [comment]
Re-open this as following is not yet done:
 [ ] Extend RetRTL to make it be able to send CMD towards controller.

But lower this priority.
